src/server/services/VirusScanService.ts

```diff
@@ -0,0 +1,77 @@
+import fetch from 'cross-fetch'
+import CrossFormData from 'form-data'
+
+import { injectable } from 'inversify'
+import { logger, virusTotalKey } from '../config'
+import { VirusScanServiceInterface } from './interfaces/VirusScanServiceInterface'
+
+const VIRUS_TOTAL_ENDPOINT = 'https://www.virustotal.com/api/v3'
+
+@injectable()
+export class VirusScanService implements VirusScanServiceInterface {
+  private static async checkResponseThenJSON(response: Response): Promise<any> {
+    const result = await response.json()
+    if (!response.ok) {
```

Comment:
should the HTTP status be checked before deserialisation?

Reply:
yes, that's what the implementation does. At any rate VirusScanService is not actually used - look at the other impl